src/main/java/com/greghaskins/spectrum/Spectrum.java

```diff
@@ -371,7 +414,16 @@ private static Suite getCurrentSuiteBeingDeclared() {
     return getSuiteStack().peek();
   }
 
-  private static void addHook(HookContext hook) {
-    getCurrentSuiteBeingDeclared().addHook(hook);
+  /**
+   * If the definition is at the right level and the block is the right sort
+   * of block, then this is might be an opportunity to hook in JUnit rules.
+   * @param definitionBlock the block used to make this suite
+   */
+  private static void addRootJUnitHook(com.greghaskins.spectrum.Block definitionBlock) {
```

Comment:
Maybe play with this implementation a little bit. It seems like we know declaratively when we are in the root block (in the `Spectrum` constructor), so the `if` statements and type cast should in theory not be needed.